src/Extensions/SiteTreeSubsites.php

```diff
@@ -398,8 +398,12 @@ public static function contentcontrollerInit($controller)
             SSViewer::set_themes(ThemeResolver::singleton()->getThemeList($subsite));
         }
 
-        if ($subsite && i18n::getData()->validate($subsite->Language)) {
-            i18n::set_locale($subsite->Language);
+        $ignore_subsite_locale = Config::inst()->get(self::class, 'ignore_subsite_locale');
+
+        if ($subsite && $subsite->Language) {
```

Comment:
Why was this added? We're already checking `$subsite` for truthiness if the below condition so all this adds is a check on truthiness for `$subsite->Language` - is there a chance this doesn't have a value?
If so, please just add it to the condition below. Nested conditions like this are a bit of an anti-pattern.

Reply:
Had the problem of $subsite existing and $subsite->Language had no value. When no subsite is defined in cms this throws a Notice. I moved this to the if below now.